indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -120,6 +147,18 @@ public Response specPost(final SupervisorSpec spec, @Context final HttpServletRe
     );
   }
 
+  private Set<ResourceAction> getNeededResourceActionsForTask(final SupervisorSpec spec)
+  {
+    final Set<ResourceAction> resourceActions = new HashSet<>();
+    resourceActions.addAll(spec.getDataSources().stream()
```

Comment:
Static analysis is failing with the `addAll` call. Think it can be done inline with the stream() chain:
```java
final Set<ResourceAction> resourceActions = spec.getDataSources().stream()
                                                    .map(dataSource -> new ResourceAction(new Resource(
                                                        dataSource,
                                                        ResourceType.DATASOURCE
                                                    ), Action.WRITE)).collect(Collectors.toSet());
```